### DISCUSSION TITLE
How to add arguments that has the form of [arg1, arg2, ...] ?

### DISCUSSION THREAD

**Discussion by citystrawman** [Tyrrrz/CliWrap#231] [Q&A]
Hi, I am using CliWrap to call exe within C#. As for my exe, there's arguments like:
`antsRegistration --output [ E:\Dir\To\Output\LHoutput, E:\Dir\To\Output\LHoutputWarped.nii.gz,E:\Dir\To\Output\LHoutputInverseWarped.nii.gz ] `

I tried to formulate the arguments that is in the square brackets in three ways, none of it has same effect as the above:
(outputPrefix, outputWarp and outputInverse are strings corresponding to E:\Dir\To\Output\LHoutput,  E:\Dir\To\Output\LHoutputWarped.nii.gz and E:\Dir\To\Output\LHoutputInverseWarped.nii.gz)

method 1:
```
var result2 = await Cli.Wrap(registrationPath)
.WithArguments(args => args
.Add("--output")
.Add([outputPrefix, outputWarp, outputInverse ])
).ExecuteBufferedAsync();
```
the arguments has no square brackets(also missing comma): 
![image](https://github.com/Tyrrrz/CliWrap/assets/9004748/76f32e2b-6ed1-4c50-a668-49e9eb4e4fb7)

method 2:
```
var result2 = await Cli.Wrap(registrationPath)
.WithArguments(args => args
.Add("--output")
.Add($"[{outputPrefix}, {outputWarp}, {outputInverse} ]")
).ExecuteBufferedAsync();
```
the arguments has extra quotes:
![image](https://github.com/Tyrrrz/CliWrap/assets/9004748/77ca4bb0-7a9f-425f-acce-85f8d8a67fe0)

method3: 
```
var result2 = await Cli.Wrap(registrationPath)
.WithArguments(args => args
.Add("--output")
.Add(new [ ]{outputPrefix, outputWarp, outputInverse} )
).ExecuteBufferedAsync();
```
the arguments are same as method 1:
![image](https://github.com/Tyrrrz/CliWrap/assets/9004748/28f31d59-4cb2-482e-a9af-5734c1d03041)

I appreciate if anyone could help me solve this issue. Thank you!

**Comment by Tyrrrz** [comment]
That's a *very* unusual command line format. In order to satisfy, you will probably need to write something like this:

```csharp
var result = await Cli.Wrap(registrationPath)
    .WithArguments(args => args
        .Add("--output")
        .Add(["[", outputPrefix, ",", outputWarp, ",", outputInverse, "]"])
).ExecuteBufferedAsync();
```

Or:

```csharp
var result = await Cli.Wrap(registrationPath)
    .WithArguments(["--output", "[", outputPrefix, ",", outputWarp, ",", outputInverse, "]"])
).ExecuteBufferedAsync();
```

Or adapted from your `method2` (not recommended):

```csharp
var result2 = await Cli.Wrap(registrationPath)
.WithArguments(args => args
.Add("--output")
.Add($"[{outputPrefix}, {outputWarp}, {outputInverse} ]", false)
).ExecuteBufferedAsync();
```

**Comment by citystrawman** [comment]
> That's a _very_ unusual command line format. In order to satisfy, you will probably need to write something like this:
> 
> ```cs
> var result = await Cli.Wrap(registrationPath)
>     .WithArguments(args => args
>         .Add("--output")
>         .Add(["[", outputPrefix, ",", outputWarp, ",", outputInverse, "]"])
> ).ExecuteBufferedAsync();
> ```
> 
> Or:
> 
> ```cs
> var result = await Cli.Wrap(registrationPath)
>     .WithArguments(["--output", "[", outputPrefix, ",", outputWarp, ",", outputInverse, "]"])
> ).ExecuteBufferedAsync();
> ```
> 
> Or adapted from your `method2` (not recommended):
> 
> ```cs
> var result2 = await Cli.Wrap(registrationPath)
> .WithArguments(args => args
> .Add("--output")
> .Add($"[{outputPrefix}, {outputWarp}, {outputInverse} ]", false)
> ).ExecuteBufferedAsync();
> ```

Thank you! I just found that the it is not the double quote that makes the error, it is that the space between square brackets and the string of directory that makes the error. If I delete the space between square brackets and the string of directory it works fine.
